src/resources/pipe_win.cc

```diff
@@ -267,17 +279,50 @@ PRIMITIVE(create_pipe) {
   resource_proxy->set_external_address(pipe_resource);
 
   array->at_put(0, resource_proxy);
+  // Windows handles are actually limited to 24 bit so this should work
+  // OK.
   array->at_put(1, Smi::from(reinterpret_cast<word>(input ? read : write)));
 
   return array;
 }
 
+class CopyPipeState {
+ public:
+  CopyPipeState(HANDLE from, HANDLE to) : from_(from), to_(to) {}
+
+  DWORD copy_loop() {
+    char buffer[4096];
+    DWORD read_count;
+    DWORD write_count;
+    while (ReadFile(from_, buffer, sizeof(buffer), &read_count, NULL) && read_count > 0) {
+      if (!WriteFile(to_, buffer, read_count, &write_count, NULL)) {
+        return 1;
+      }
+    }
+    CloseHandle(from_);
+    CloseHandle(to_);
+    delete(this);
```

Comment:
I'd add a comment here that explains why this is a good idea (I don't really think it is).

```suggestion
    delete this;
```